scripts/monthly_means.py

```diff
@@ -0,0 +1,242 @@
+    if product == 'gridsat':
+        time_deltas = [i * np.timedelta64(3, 'h') for i in range(8)]
+    else:
+        time_deltas = [i * np.timedelta64(30, 'm') for i in range(24 * 2)]
+    
+    # .astype('datetime64[M]').astype('datetime64[m]'):
+    # hack to avoid dealing with days, i.e. date set to YYYYmmddT00:00
+    time_offset = ds.time.values[0].astype('datetime64[M]').astype('datetime64[m]')
+    time_values = [time_offset + delta for delta in time_deltas]
+    
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message=(
+            "Converting non-nanosecond precision datetime "
+            "values to nanosecond precision."
+            )
+        )
+        # Floor day to first day of the month
+        ds['time'] = ds['time'] - np.array(
+            [np.timedelta64(d - 1, 'D') for d in ds.time.dt.day.values]
+        )
+        ds = ds.reindex({'time': time_values}, method=None, fill_value=0)
+
+    # Initialize all variables to zero and create a count variable
+    variables = set(ds.variables) - set(ds.coords)
+    for v in variables:
+        # float instead of float32 to avoid any limitations accumulating
+        # .copy to assign coordinates correctly
+        ds[v] = ds[v].copy(
+            data=np.zeros_like(ds[v]), deep=True
+        ).astype(float)
+        ds[f'{v}_count'] = ds[v].copy(
+            data=np.zeros_like(ds[v]), deep=True
+        ).astype(int)
+
+    # Accumulate values
+    for f in tqdm(files, dynamic_ncols=True):
+        ds_f = xr.open_zarr(f).load()
+
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", message=(
+                "Converting non-nanosecond precision"
+                )
+            )
+            # Replace the timestamps day with first day of the month
+            # and reindex to handle time dimension
+            ds_f['time'] = ds_f['time'] - np.array(
+                [np.timedelta64(d - 1, 'D') for d in ds_f.time.dt.day.values]
+            )
+            ds_f = ds_f.reindex({'time': time_values},
+                                method=None, fill_value=np.nan)
+        
+        for v in variables:
+            is_finite = np.isfinite(ds_f[v].data)
+            ds[v] = ds[v] + np.where(is_finite, ds_f[v].data, 0)
+            ds[f'{v}_count'] = ds[f'{v}_count'] + is_finite.astype(int)
+
+    # Divide by the total count
+    for v in variables:
+        non_zero_count = ds[f'{v}_count'].data > 0
+        # .copy to set dimensions correctly
+        ds[v] = ds[v].copy(
+            data=np.divide(ds[v].data, ds[f'{v}_count'].data,
+                           out=np.full_like(ds[v].data, np.nan),
+                           where=non_zero_count),
+            deep=True
+        ).astype(np.float32)
+        ds[f'{v}_count'] = ds[f'{v}_count'].astype(np.int16)
+
+    # Compute the monthly aggregates
+    ds['month'] = (('month',), [ds.time.data.min()])
+    for v in variables:
+        ds[f'{v}_aggregated'] = (
```

Comment:
I would really like to drop the ``_aggregated`` suffix. Any mean is by definitions aggregated, which is why I think this clutter.

Reply:
Updated with bf4cf7c.

Now:
```
In [5]: ds
Out[5]: 
<xarray.Dataset>
Dimensions:                         (latitude: 2, longitude: 2, time: 8,
                                     month: 1)
Coordinates:
  * latitude                        (latitude) float32 0.0 -0.07
  * longitude                       (longitude) float32 -0.1 -0.03
  * time                            (time) datetime64[ns] 2010-01-01 ... 2010...
    altitude                        float64 500.0
  * month                           (month) datetime64[ns] 2010-01-01
Data variables:
    cloud_prob_2d_stratified        (time, latitude, longitude) float32 0.028...
    inpainted_stratified            (time, latitude, longitude) float32 0.0 ....
    p_tiwp_stratified               (time, latitude, longitude) float32 0.026...
    tiwp_stratified                 (time, latitude, longitude) float32 0.0 ....
    p_tiwp_stratified_count         (time, latitude, longitude) int16 2 2 ... 0
    cloud_prob_2d_stratified_count  (time, latitude, longitude) int16 1 2 ... 0
    tiwp_stratified_count           (time, latitude, longitude) int16 2 2 ... 0
    inpainted_stratified_count      (time, latitude, longitude) int16 2 2 ... 0
    p_tiwp                          (month, latitude, longitude) float32 0.01...
    cloud_prob_2d                   (month, latitude, longitude) float32 0.00...
    tiwp                            (month, latitude, longitude) float32 0.0 ...
    inpainted                       (month, latitude, longitude) float32 0.0 ...
Attributes:
    history:          2023-12-05 10:19:19.864566: Monthly means computation
    input_filename:   ['ccic_gridsat_201001010000.zarr', 'ccic_gridsat_201001...
    institution:      Chalmers University of Technology
    processing_time:  2023-06-21T19:16:26.249990
    source:           framework: ccic-0.0
    title:            The Chalmers Cloud Ice Climatology
 ```

The coordinate `month` has dtype `datetime64[ns]`, I do not manage to get it to be `datetime64[M]`, but that's a minor issue (and maybe a limitation of xarray?)